src/pages/Search/index.tsx

```diff
@@ -1,63 +1,46 @@
-import { useSearchParams } from 'react-router-dom';
-import { useState } from 'react';
+import React, { useState } from 'react';
 import cn from 'utils/ts/classNames';
-import RollingBanner from './components/RollingBanner';
-import list from './static/data';
-import styles from './Search.module.scss';
-import Suggestion from './components/Suggestion';
-import Navigation from './components/Navigation';
-import SearchInput from './components/SearchInput';
-import Recommendation from './components/Recommendation';
-
-type CurrentMode = string | null;
-
-const useSearchForm = (state : CurrentMode) => {
+import styles from 'pages/Search/Search.module.scss';
+import Recommendation from './components/SearchBar/Recommendation';
+import SearchInput from './components/SearchBar/SearchInput';
+import RollingBanner from './components/SearchBar/RollingBanner';
+import NavigationBar from './components/NavigationBar';
+import RelatedSearches from './components/RelatedSearches';
+import useTrendingList from './hooks/useTrendings';
+import useSearchingMode from './hooks/useSearchingMode';
+
+const useSearchForm = () => {
   const [text, setText] = useState('');
-  const [mode, setMode] = useState(state);
-
   const handleText = (e : React.ChangeEvent<HTMLInputElement>) => {
     setText((e.target.value));
   };
 
-  const changeSearchMode = () => {
-    setMode('search');
-  };
-
-  const changeTrendingMode = () => {
-    setMode('trending');
-    setText('');
-  };
-
   return {
-    text, mode, handleText, changeSearchMode, changeTrendingMode,
+    text, handleText,
   };
 };
 
 export default function Search(): JSX.Element {
-  const [searchParams] = useSearchParams();
-  const currentMode : CurrentMode = searchParams.get('mode');
-  const {
-    text, mode, handleText, changeSearchMode, changeTrendingMode,
-  } = useSearchForm(currentMode || 'trending');
+  const { text, handleText } = useSearchForm();
+  const { isLoading, data } = useTrendingList();
```

Comment:
hook에서 리턴하는 값은 `data`로 일반적인 것이 좋지만, 이를 활용하는 컴포넌트에서도 일반적인 이름을 쓰는건 중복의 위험이 있어보여요.
```suggestion
  const { isLoading, data: trendings } = useTrendingList();
```

Reply:
data: trendings로 표현하는건 좋다고 생각합니다. 그런데 자식 컴포넌트의 props로 넘길때는 자식 컴포넌트에서 받는 프로퍼티 이름과 같아지게 되는데 
ex) `<RollingBanner trendings={trendings} />` 처럼 말이죠
이렇게 겹치는 경우는 상관 없는건가요? 뭔가 이름이 중복된다는 느낌이 들어서요.

Reply:
자식 컴포넌트에서 해당 훅을 호출하는 방법도 있습니다 :D